### PR TITLE
batchmark: allow resample instances and reduction of partial results

### PR DESCRIPTION
**Pull request by mllg**

### CODE REVIEW

**Comment by mllg** [comment]
@berndbischl I've changed reduceBatchmarkResults to work with subsets of jobs. I find this extremely useful to peek into results while some jobs are still running. On the other hand, I'm unsure if the returned object is still a valid BMR. Thoughts?

File: R/batchmark.R [hunk 1]
```diff
@@ -109,38 +100,37 @@ reduceBatchmarkResults = function(ids = NULL, keep.pred = TRUE, show.info = getM
...
+  setkeyv(tab, cols = c("problem", "algorithm"), physical = FALSE)
+  result = namedList(tab[, unique(problem)])
+
+  for (prob in names(result)) {
+    algos = unique(tab[problem == prob], by = "algorithm")
+    data = batchtools::makeJob(id = algos$job.id[1L])$problem$data
+    result[[prob]] = namedList(algos$algorithm)
+
+    for (algo in names(result[[prob]])) {
+      res = batchtools::reduceResultsList(tab[problem == prob & algorithm == algo])
```
Comment (pfistfl) [inline R/batchmark.R]: Something I stumbled upon while playing around:
Why is `, reg = reg` not passed here? 
As I see it for now it only works with default registries?

Comment (mllg) [inline R/batchmark.R]: Right, I've fixed this in 2 lines.

File: R/batchmark.R [hunk 2]
```diff
@@ -36,43 +36,34 @@
...
 
   reg$packages = union(reg$packages, "mlr")
 
   # generate problems
-  pdes = Map(function(id, task, rdesc, seed) {
-    batchtools::addProblem(id, data = list(rdesc = rdesc, task = task, measures = measures, learners = learners), fun = resample.fun, seed = seed, reg = reg)
-    data.table(i = seq_len(rdesc$iters))
-  }, id = task.ids, task = tasks, rdesc = resamplings, seed = reg$seed + seq_along(tasks))
+  pdes = Map(function(id, task, rin, seed) {
+    batchtools::addProblem(id, data = list(rin = rin, task = task, measures = measures, learners = learners), fun = resample.fun, seed = seed, reg = reg)
```
Comment (pfistfl) [inline R/batchmark.R]: ``reg = reg`` is additionally missing in  Lines 50

**Comment by pfistfl** [comment]
Hi, 

found another bug (I think the following code should work:
```
library(mlr)

reg = makeExperimentRegistry("../reg/")
tsks = iris.task
lrns = list("classif.rpart", "classif.lda")
batchmark(lrns, tsks, cv2, acc)

# Run only first two jobs
t = getJobTable(reg = reg)
submitJobs(id = t$job.id[1:2], reg = reg)
waitForJobs(id = t$job.id[1:2], reg = reg)

reduceBatchmarkResults(ids = t$job.id[1:2], reg = reg)
```

> [Resample] Aggr. Result: acc.test.mean=0.94
> Error in res[[1L]] : subscript out of bounds

Should I create a PR on top of yours?

**Comment by pfistfl** [comment]
EDIT: Sorry, was in a wrong branch :(
It actually works as expected.

~~found another bug (I think the following code should work :~~
```
library(mlr)

reg = makeExperimentRegistry("../reg/")
tsks = iris.task
lrns = list("classif.rpart", "classif.lda")
batchmark(lrns, tsks, cv2, acc)

# Run only first two jobs
t = getJobTable(reg = reg)
submitJobs(id = t$job.id[1:2], reg = reg)
waitForJobs(id = t$job.id[1:2], reg = reg)

reduceBatchmarkResults(ids = t$job.id[1:2], reg = reg)
```

~~ >> [Resample] Aggr. Result: acc.test.mean=0.94
~~ >> Error in res[[1L]] : subscript out of bounds

**Comment by larskotthoff** [comment]
Merging. If anything needs to be added, please open a new PR.

**Pull request merged**